### PR TITLE
Fix minor bugs in add_parent_mass

### PR DESCRIPTION
**Pull request by florian-huber**
Addressing #185 and #180

- moved print statement to correct position in code
- added tests for missing cases
- test that returned parent_mass is float

### CODE REVIEW

File: matchms/filtering/add_parent_mass.py [hunk 1]
```diff
@@ -36,11 +36,11 @@ def add_parent_mass(spectrum_in: SpectrumType, estimate_from_adduct: bool = True
...
             precursor_mz = spectrum.get("precursor_mz", None)
             if precursor_mz is None:
-                precursor_mz = spectrum.get("pepmass")[0]
-        except KeyError:
+                precursor_mz = spectrum.get("pepmass", None)[0]
+        except TypeError:
             print("Not sufficient spectrum metadata to derive parent mass.")
+            return spectrum
 
-        spectrum = spectrum_in.clone()
```
Comment (sverhoeven) [inline matchms/filtering/add_parent_mass.py:43]: Good catch. cloning twice is wasteful

File: tests/test_add_parent_mass.py [hunk 2]
```diff
@@ -17,9 +17,24 @@ def test_add_parent_mass():
...
+    mz = numpy.array([], dtype='float')
+    intensities = numpy.array([], dtype='float')
+    metadata = {"charge": -1}
+    spectrum_in = Spectrum(mz=mz,
+                           intensities=intensities,
+                           metadata=metadata)
+
+    spectrum = add_parent_mass(spectrum_in)
+
+    assert spectrum.get("parent_mass") is None, "Expected no parent mass"
```
Comment (sverhoeven) [inline tests/test_add_parent_mass.py]: Also assert if print happened by using https://docs.pytest.org/en/stable/capture.html
```suggestion
def test_add_parent_mass_no_pepmass(capsys):
    """Test if correct expection is returned."""
    mz = numpy.array([], dtype='float')
    intensities = numpy.array([], dtype='float')
    metadata = {"charge": -1}
    spectrum_in = Spectrum(mz=mz,
                           intensities=intensities,
                           metadata=metadata)

    spectrum = add_parent_mass(spectrum_in)

    assert spectrum.get("parent_mass") is None, "Expected no parent mass"
    assert "Not sufficient spectrum metadata to derive parent mass." in capsys.readouterr().out
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Fixed
+
+- Fix very minor bugs in `add_parent_mass` [#188](https://github.com/matchms/matchms/pull/188)
```
Comment (sverhoeven) [inline CHANGELOG.md]: Dont repeat heading
```suggestion
- Very minor bugs in `add_parent_mass` [#188](https://github.com/matchms/matchms/pull/188)
```

**Review by sverhoeven — CHANGES_REQUESTED** [review]
Changes works as expected. Good coverage from tests.

To prevent #180 from popping up again could you add assertions for the `print()`? 
See inline comment on `test_add_parent_mass_no_pepmass` for example.

File: tests/test_add_parent_mass.py (hunk 2 above)
Comment (florian-huber) [reply tests/test_add_parent_mass.py]: Nice, thanks! Didn't know how to handle that before, so glad to see how that works.

**Comment by florian-huber** [comment]
I added more tests of expected presence/absence of print statements to the unit tests (not to every test, but too all tests where it did play a role before this PR).

**Review by sverhoeven — APPROVED** [review]
Good work.

File: tests/test_add_parent_mass.py [hunk 4]
```diff
@@ -51,6 +70,23 @@ def test_add_parent_mass_using_adduct(adduct, expected):
...
+    metadata = {"precursor_mz": 444.0}
+    spectrum_in = Spectrum(mz=mz,
+                           intensities=intensities,
+                           metadata=metadata)
+
+    spectrum = add_parent_mass(spectrum_in)
+
+    assert spectrum.get("parent_mass") is None, "Expected no parent mass"
+    assert "Not sufficient spectrum metadata to derive parent mass." in capsys.readouterr().out
+
```
Comment (sverhoeven) [inline tests/test_add_parent_mass.py:89]: Make linter happy
```suggestion
```

File: tests/test_add_parent_mass.py (hunk 2 above)
Comment (sverhoeven) [reply tests/test_add_parent_mass.py]: Just learned it myself, needed it for howfairis last week. New entry for @felipeZ [cookbook](https://blog.esciencecenter.nl/a-quick-and-dirty-pytest-cookbook-58051a2980c3)?

**Comment by florian-huber** [comment]
Thanks a lot for reviewing Stefan!

**Comment by felipeZ** [comment]
@sverhoeven  thanks for the hint about `capsys`, I am going to include it in the cookbook!

**Pull request merged**